### PR TITLE
Call `require_grad()` on ChainerX `Variable.grad` setter 

### LINKED ISSUE #7190: Make `Variable.grad.setter` call `arr.require_grad()` for ChainerX array in order to align the double-backprop behavior with NumPy/CuPy
Original issue: The following code fails at the 2 commented lines. It succeeds for `numpy` and `cupy`.

```py
import chainerx as xp
import chainer
from chainer import functions as F


x_arr = xp.array([1, 2], 'float32')
gy_arr = xp.array([3, 4], 'float32')
ggx_arr = xp.array([5, 6], 'float32')

x = chainer.Variable(x_arr, requires_grad=True)
y = F.identity(x)
y.grad = gy_arr

gy = y.grad_var
y.backward(enable_double_backprop=True)


assert x.grad_var.requires_grad is True  # fail
x.grad_var.grad = ggx_arr  # error
x.grad_var.backward()


print(gy.grad_var)
```

### PR DESCRIPTION
**Pull request by niboshi**
Fixes #7190

~Depends on #7199~

### CODE REVIEW

**Comment by hvy** [comment]
Hm, BN tests are failing. 🤔

File: chainer/variable.py [hunk 1]
```diff
@@ -925,6 +925,9 @@ def _set_chainerx_grad(self, g):
                 raise RuntimeError(
                     'Cannot set a gradient to an empty variable')
         elif arr.is_backprop_required():
+            # If g is grad-stopped, require grad on it.
+            if g is not None and not g.is_backprop_required():
+                g.require_grad()
```
Comment (hvy) [inline chainer/variable.py]: Discussed offline to create a view of `g` so that the user's gradient isn't modified. This is to keep similar semantics to the array passed as the data to `Variable.__init__`.

File: chainer/variable.py [hunk 2]
```diff
@@ -925,6 +925,10 @@ def _set_chainerx_grad(self, g):
                 raise RuntimeError(
                     'Cannot set a gradient to an empty variable')
         elif arr.is_backprop_required():
+            # If g is grad-stopped, require grad on it.
+            # Make a view in order not to affect the input.
+            if g is not None and not g.is_backprop_required():
+                g = g.view().require_grad()
```
Comment (hvy) [inline chainer/variable.py]: `_set_chainerx_grad` is called from the `grad_var` setter too. If the argument variable to a call to it holds a `chainerx.ndarray` which doesn't require any gradients, it will be modified to require gradients here. Maybe pass a third flag argument to this method to control it?

```python
import chainerx as chx
import chainer as ch
v = ch.Variable(chx.ones(3, dtype='f'), requires_grad=True)
g = ch.Variable(chx.ones(3, dtype='f'), requires_grad=False)
v.grad_var = g
v.grad_var.requires_grad  # `True`, but should be `False`. It'll be `False` if `chx` is replaced with e.g. `numpy`.

```

File: tests/chainer_tests/test_variable.py [hunk 3]
```diff
@@ -757,30 +757,57 @@ def test_copydata_from_to_uninitialized_parameters(
...
+            backend_config.xp.testing.assert_array_equal(v.grad, g)
 
     def test_grad_var(self, backend_config):
         x = backend_config.get_array(
             np.random.uniform(-1, 1, self.shape).astype(np.float32))
         g = backend_config.get_array(
             np.random.uniform(0.1, 10, self.shape).astype(np.float32))
-        v = chainer.Variable(x)
+        v = chainer.Variable(x, requires_grad=self.requires_grad)
         gv = chainer.Variable(g)
```
Comment (hvy) [inline tests/chainer_tests/test_variable.py]: Can you parameterize `requires_grad` for `gv` and check its state too?

File: chainer/variable.py (hunk 2 above)
Comment (niboshi) [reply chainer/variable.py]: Fixed.

File: tests/chainer_tests/test_variable.py (hunk 3 above)
Comment (niboshi) [reply tests/chainer_tests/test_variable.py]: Fixed.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
LGTM!

**Comment by hvy** [comment]
LGTM, failure seems unrelated.
